sonar-bukkit/src/main/java/xyz/jonesdev/sonar/bukkit/fallback/FallbackBukkitInboundHandler.java

```diff
@@ -121,9 +122,8 @@ public void channelRead(final @NotNull ChannelHandlerContext ctx, final @NotNull
         handleLogin(ctx.channel(), ctx, () -> {
           byteBuf.readerIndex(originalReaderIndex);
           ctx.fireChannelRead(byteBuf);
+          ctx.channel().pipeline().remove(FallbackPipelines.FALLBACK_INBOUND_HANDLER);
         }, loginStart.getUsername(), socketAddress);
-        // Release the ByteBuf to avoid memory leaks
-        byteBuf.release();
```

Comment:
We're releasing ByteBufs in this handler multiple times. Are you sure that this won't cause memory leaks?

Reply:
I found no memory leaks in my tests. The reason I removed it was that there might be a retain reference or smth in the FallbackInboundHandler.

> This was removed because there were of duplicate releases after removing FallbackInboundHandler

No problems were found in the whole process of testing with parameter `-Dio.netty.leakDetection.level=PARANOID`. But it's not certain if edge cases will happen.